solutions/components-in-depth/src/app/timer/timer.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core'
+
+const MS_IN_SECOND = 1000;
+
+@Component({
+  selector: 'app-timer',
+  templateUrl: './timer.component.html',
+  styleUrls: ['./timer.component.css']
+})
+export class TimerComponent {
+  @Input() interval: number = 50;
+  @Output() onInterval = new EventEmitter();
+  @Output() onReset = new EventEmitter();
+  @Output() onStop = new EventEmitter();
+  private time: number = 0;
+  private timerId: NodeJS.Timer;
+
+  start(): void {
+    if (this.timerId) {
+      return;
+    }
+
+    this.timerId = setInterval(() => {
+      this.time = this.computeTime(this.time, this.interval);
+      this.onInterval.next(this.time);
```

Comment:
It looks like `next` is the old and deprecated version of `emit`

http://stackoverflow.com/questions/35840576/difference-between-eventemitter-next-and-eventemitter-emit-in-angular-2